utils/datasets/cifar10_dataset_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019, Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+
+import tensorflow as tf
+
+from utils.datasets import cifar10_dataset
+
+
+def _compute_length_of_dataset(ds):
+  return ds.reduce(0, lambda x, _: x + 1)
+
+
+class DatasetTest(tf.test.TestCase):
```

Comment:
It might be worth having a test to ensure that the number of clients is 10, and that (at least one of) the clients have the correct number of batches.